src/shared/all.sh

```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
+
+CICD_TOOLS_COMMON_LOADED=1
+CICD_TOOLS_CONTAINER_ENGINE_LOADED=1
+CICD_TOOLS_WORKDIR=${CICD_TOOLS_WORKDIR:-$(pwd)}
+
+source "${CICD_TOOLS_WORKDIR}/src/shared/common.sh"
```

Comment:
What is the purpose of `all.sh` vs `bootstrap.sh` -- currently they appear to source the same two scripts but I guess in the future you envision that they might load different scripts?

Reply:
Yeah - My intention here is that `bootstrap` takes an argument to decide which functions to load. 

For now, it accepts `all` and `container engine` - and by default, it uses `all`.

I want to follow up this PR with another adding more helpers with different contexts, the main idea being one being able to select only which dependencies their script needs.

Reply:
I've added a README section to explain how to use it, maybe it is clearer now - for the moment it only has the container lib though.